indexer-opts/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+    pub start_mode: StartMode,
+    /// Database URL
+    #[clap(long, short, env)]
+    pub database_url: String,
+}
+
+#[derive(ArgEnum, Debug, Clone, PartialEq, Eq)]
+pub enum ChainId {
+    Mainnet,
+    Testnet,
+}
+
+#[derive(ArgEnum, Debug, Clone)]
+pub enum StartMode {
+    FromLatest,
+    FromInterruption,
+}
+
+#[derive(Debug)]
+pub enum MetaAction {
+    RegisterIndexer {
+        indexer_id: String,
+        indexer_type: String,
+        start_block_height: u64,
+        end_block_height: Option<u64>,
+    },
+    UpdateMeta {
+        indexer_id: String,
+        last_processed_block_height: u64,
+    },
+}
+
+pub async fn update_meta(
+    db_with_meta_data_pool: &sqlx::Pool<sqlx::Postgres>,
+    action: MetaAction,
+) -> anyhow::Result<()> {
+    match action {
+        MetaAction::UpdateMeta {
+            indexer_id,
+            last_processed_block_height,
+        } => {
+            let block_height: bigdecimal::BigDecimal =
+                match bigdecimal::BigDecimal::from_u64(last_processed_block_height) {
+                    Some(value) => value,
+                    None => anyhow::bail!("Failed to parse u64 to BigDecimal"),
+                };
+            match sqlx::query!(
+                r#"
+UPDATE __meta SET last_processed_block_height = $1 WHERE indexer_id = $2
+                "#,
+                block_height,
+                indexer_id,
+            )
+            .execute(db_with_meta_data_pool)
+            .await
+            {
+                Ok(_) => Ok(()),
+                Err(err) => {
+                    tracing::warn!(
+                        "Failed to update meta for INDEXER ID {}\n{:#?}",
+                        indexer_id,
+                        err,
+                    );
+                    anyhow::bail!(err)
+                }
+            }
+        }
+        MetaAction::RegisterIndexer {
+            indexer_id,
+            indexer_type,
+            start_block_height,
+            end_block_height,
+        } => {
+            apply_migration(db_with_meta_data_pool).await?;
+            let block_height: bigdecimal::BigDecimal =
+                match bigdecimal::BigDecimal::from_u64(start_block_height) {
+                    Some(value) => value,
+                    None => anyhow::bail!("Failed to parse u64 to BigDecimal"),
+                };
+            let end_block_height = if let Some(end_block_height) = end_block_height {
+                bigdecimal::BigDecimal::from_u64(end_block_height)
+            } else {
+                None
+            };
+
+            sqlx::query!(
+                r#"
+INSERT INTO __meta (indexer_id, indexer_type, indexer_started_at, last_processed_block_height, start_block_height, end_block_height)
+VALUES ($1, $2, now(), $3, $3, $4)
+ON CONFLICT (indexer_id) DO UPDATE
+    SET start_block_height = EXCLUDED.start_block_height,
+        last_processed_block_height = EXCLUDED.start_block_height,
```

Comment:
```suggestion
```

We kill all the feature with this line, we don't need to update `last_processed_block_height` here